File: 1910-CheckIfBinaryStringHasAtMostOneSegmentOfOnes/1910-CheckIfBinaryStringHasAtMostOneSegmentOfOnes.cpp
// Last updated: 3/30/2026, 2:11:52 PM
class Solution {
public:
    bool checkOnesSegment(string s) {
        for(int index = 1 ; index < s.size() ; index++){
            if(s[index] == '1' && s[index-1] == '0') return false;
        }
        return true;
    }
};

File: 1910-CheckIfBinaryStringHasAtMostOneSegmentOfOnes/1910-CheckIfBinaryStringHasAtMostOneSegmentOfOnes.java
// Last updated: 3/30/2026, 2:11:56 PM
class Solution {
    public boolean checkOnesSegment(String s) {
        char[] arr = s.toCharArray();
        for(int index = 1 ; index < s.length() ; index++){
            if(arr[index] == '1' && arr[index-1] == '0'){
                return false;
            }
        }
        return true;
    }
}